### PR TITLE
Time accessor

### LINKED ISSUE #6: Support groupby's datetime accessor behaviour
```
ds.cf.groupby("T.month") → ds.groupby("time.month")
```

### PR DESCRIPTION
**Pull request by jukent**
Closes #6

### CODE REVIEW

**Comment by jukent** [comment]
@dcherian Sorry the pullbot is very destructive! If I ever forget to create the PR from a separate branch it closes it on me.

**Comment by jukent** [comment]
I am still developing my skills at reading failed test reports, but from what I can gather the test fails in `test_wrapped_classes`. The error is `KeyError: 'T.month'` from a line in the xarray/dataset library (`/usr/share/miniconda/envs/cf_xarray_test/lib/python3.6/site-packages/xarray/core/dataset.py:1166: KeyError`). 

What I am struggling to figure out is at what line in `test_wrapped_classes` we call up dataset. I suppose it is in the `getattr` lines where we establish the expected and actual datasets (line 159 or 160).

I'd appreciate any help with this.

File: cf_xarray/accessor.py [hunk 1]
```diff
@@ -308,7 +338,7 @@ def _get_measure(da: Union[DataArray, Dataset], key: str) -> List[str]:
     "dims_or_levels": (_get_axis_coord,),  # reset_index
     "window": (_get_axis_coord,),  # rolling_exp
     "coord": (_get_axis_coord_single,),  # differentiate, integrate
-    "group": (_get_axis_coord_single,),
+    "group": (_get_axis_coord_single, _get_axis_coord_time_accessor),
```
Comment (dcherian) [inline cf_xarray/accessor.py:341]: missed a step ;)

File: cf_xarray/tests/test_accessor.py [hunk 2]
```diff
@@ -152,7 +153,7 @@ def test_wrapped_classes(obj, attr, xrkwargs, cfkwargs):
         actual = getattr(obj.cf, attr)(**cfkwargs).mean(*args)
     assert_identical(expected, actual)
 
-    if attr in ("groupby", "groupby_bins"):
+    if attr in ("groupby", "groupby_time_accessor", "groupby_bins"):
```
Comment (dcherian) [inline cf_xarray/tests/test_accessor.py:156]:
```suggestion
    if attr in ("groupby", "groupby_bins"):
```

**Comment by dcherian** [comment]
I forgot to say, the `KeyError: T.month` told me your function wasn't being applied so I tried it in the example notebook and debugged from there.

File: cf_xarray/accessor.py (hunk 1 above)
Comment (jukent) [reply cf_xarray/accessor.py:341]: No `,` after `_get_axis_coord_time_accessor` ?

File: cf_xarray/tests/test_accessor.py (hunk 2 above)
Comment (jukent) [reply cf_xarray/tests/test_accessor.py:156]: I have not made this change yet, as I'm a little confused by one comment adding "groupby_time_accessor" and one removing it. Currently it IS IN the code and it is passing.

Comment (dcherian) [reply cf_xarray/tests/test_accessor.py:156]: yes we should delete it because it isn't doing anything.

File: cf_xarray/accessor.py (hunk 1 above)
Comment (dcherian) [reply cf_xarray/accessor.py:341]: I added `_get_axis_coord_time_accessor`. It was missing. I think your branch was a little messed up after the `pull` bot did its thing.

**Comment by dcherian** [comment]
Thanks @jukent  this is great!

**Pull request merged**